Fix #1796: Cleanup the library

sjrd opened the pull request:
Some things that were *not* done, and that we could consider in the future:

* Privatize `StackTrace.JSStackTraceElem`, because it is referenced by `EnvironmentInfo`.
* Move `annotation.JavaDefaultMethod` to `internal._`. Technically it is not internal, as it is meant to be used in user source code. But its legit uses are really rare, and must typically done with extreme care, so we may not want to keep that annotation in plain sight.

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4029/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4029/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5339/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5339/</a>
Test FAILed.

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4030/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4030/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5340/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5340/</a>
Test PASSed.

gzm0 commented on library/src/main/scala/scala/scalajs/js/JSNumberOps.scala:
```diff
@@ -93,13 +93,13 @@ object JSNumberOps {
   @deprecated("A Long is converted to Double to perform JavaScript "+
       "operations. This is almost certainly not what you want. "+
-      "Use `.toDouble` explicitly if you need it.", "0.6.0")
+      "Use `.toDouble` explicitly if you need it.", "forever")
```
Maybe comment in the commit message on why we are keeping these (just like in the other commit)? Or even move these changes to the other commit.

gzm0 commented on library/src/main/scala/scala/scalajs/js/WrappedDictionary.scala:
```diff
@@ -57,7 +57,7 @@ class WrappedDictionary[A](val dict: Dictionary[A])
   }
 
   override def update(key: String, value: A): Unit =
```
Shouldn't this be `@inline`?

sjrd replied:
No, the class itself is `@inline`. If the instance of `WrappedDictionary` was not virtualized away, there's no point in inlining `update`. But if it is, then `update` will be inlined automatically.

gzm0 replied:
Right :)

gzm0 replied:
(I think I ask this every time you touch this code, don't I :P)

sjrd replied:
If you do, I forget from one time to the next ^^

gzm0 commented on library/src/main/scala/scala/scalajs/js/PropertyDescriptor.scala:
```diff
@@ -10,12 +10,21 @@
 package scala.scalajs.js
 
-@native
+import scala.scalajs.js
```
Sure about this? I think if we do this, we should do it consistently.

sjrd replied:
We've slowly been adopting this style in new code, yes. At some point we'll need a pass over the entire package to do it consistently, indeed.

sjrd replied:
I can add an additional commit with that change everywhere. It kind of fits the theme of the PR.

gzm0 replied:
I think that would be nice.

sjrd commented:
Updated. The last two commits are new. And I moved the `"0.6.x"` -> `"forever"` change to the commit that deprecates other `Long` to JS number conversions.

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4037/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4037/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5348/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5348/</a>
Test PASSed.

gzm0 commented:
Ah, this needs a rebase after #2800. I guess it is fine by you that I merged that one first ;)

sjrd commented:
> I guess it is fine by you that I merged that one first ;)

Yes, I think that was the better order :)

sjrd commented:
Rebased.

scala-jenkins commented:
Refer to this link for build results (access rights to CI server needed): 

<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4041/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-pr/4041/</a>
<a href='https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5352/'>https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/5352/</a>
Test PASSed.
